### PR TITLE
rachel-L1221.ghg_agr_R_S_T_Y_IRR

### PR DESCRIPTION
**Pull request by rhoesly**
L1221.ghg_agr_R_S_T_Y_IRR

#125

### CODE REVIEW

File: R/zchunk_L1221.ghg_agr_R_S_T_Y_IRR.R [hunk 1]
```diff
@@ -26,42 +27,29 @@ module_emissions_L1221.ghg_agr_R_S_T_Y_IRR_DISABLED <- function(command, ...) {
...
+      add_units("Tg") %>%
+      add_comments("outputs calculated in L1211.ag_irrShare_R_C_Y_GLU_irr, just add column IRR_Rfd and rearrange") %>%
       add_legacy_name("L1221.ghg_tg_R_agr_C_Y_GLU_IRR") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_NO_TEST, FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
-      L1221.ghg_tg_R_agr_C_Y_GLU_IRR
+      add_precursors("L1211.ag_irrShare_R_C_Y_GLU_irr",
+                     "L122.ghg_tg_R_agr_C_Y_GLU") %>%
+      add_flags(FLAG_NO_TEST, FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR)
```
Comment (bpbond) [inline R/zchunk_L1221.ghg_agr_R_S_T_Y_IRR.R]: Remove FLAG_NO_TEST

File: R/zchunk_L1221.ghg_agr_R_S_T_Y_IRR.R [hunk 2]
```diff
@@ -26,42 +27,29 @@ module_emissions_L1221.ghg_agr_R_S_T_Y_IRR_DISABLED <- function(command, ...) {
...
-    # }
-    #
-    #
-    # NOTE: there are `merge` and/or 'match' calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Merge-and-Match
-    # ===================================================
+    # Production shares were computed in a prior file and written out, so just rearrange in one step
+    L1221.ghg_tg_R_agr_C_Y_GLU_IRR <- L122.ghg_tg_R_agr_C_Y_GLU %>%
+      select(-value) %>%
+      repeat_add_columns( tibble(Irr_Rfd = c( "IRR", "RFD" ) ) ) %>%
```
Comment (bpbond) [inline R/zchunk_L1221.ghg_agr_R_S_T_Y_IRR.R]: No GCAM-style parenthesis spacing please

File: R/zchunk_L1221.ghg_agr_R_S_T_Y_IRR.R [hunk 3]
```diff
@@ -26,42 +27,29 @@ module_emissions_L1221.ghg_agr_R_S_T_Y_IRR_DISABLED <- function(command, ...) {
...
+      left_join_error_no_match(L1211.ag_irrShare_R_C_Y_GLU_irr, by = c("GCAM_region_ID","GCAM_commodity","GLU","Irr_Rfd","year"))
```
Comment (bpbond) [inline R/zchunk_L1221.ghg_agr_R_S_T_Y_IRR.R]: Space after commas

File: R/zchunk_L1221.ghg_agr_R_S_T_Y_IRR.R [hunk 4]
```diff
@@ -1,23 +1,24 @@
...
 #'
 #' @param command API command to execute
 #' @param ... other optional parameters, depending on command
 #' @return Depends on \code{command}: either a vector of required inputs,
 #' a vector of output names, or (if \code{command} is "MAKE") all
 #' the generated outputs: \code{L1221.ghg_tg_R_agr_C_Y_GLU_IRR}. The corresponding file in the
 #' original data system was \code{L1221.ghg_agr_R_S_T_Y_IRR.R} (emissions level1).
-#' @details Describe in detail what this chunk does.
+#' @details Downscales emissions to irrigated/rainfed technologies. These are already calculated
+#' in L1211.ag_irrShare_R_C_Y_GLU_irr
```
Comment (bpbond) [inline R/zchunk_L1221.ghg_agr_R_S_T_Y_IRR.R]: This is incorrect. Fixing.

File: R/zchunk_L1221.ghg_agr_R_S_T_Y_IRR.R [hunk 5]
```diff
@@ -1,23 +1,23 @@
...
+#' Downscale emissions to irrigated/rainfed technologies on the basis of production share.
 #'
 #' @param command API command to execute
 #' @param ... other optional parameters, depending on command
 #' @return Depends on \code{command}: either a vector of required inputs,
 #' a vector of output names, or (if \code{command} is "MAKE") all
 #' the generated outputs: \code{L1221.ghg_tg_R_agr_C_Y_GLU_IRR}. The corresponding file in the
 #' original data system was \code{L1221.ghg_agr_R_S_T_Y_IRR.R} (emissions level1).
-#' @details Describe in detail what this chunk does.
+#' @details Downscales emissions to irrigated/rainfed technologies. These are already calculated
```
Comment (cahartin) [inline R/zchunk_L1221.ghg_agr_R_S_T_Y_IRR.R]: what was already calculated?

File: R/zchunk_L1221.ghg_agr_R_S_T_Y_IRR.R [hunk 6]
```diff
@@ -26,42 +26,29 @@ module_emissions_L1221.ghg_agr_R_S_T_Y_IRR_DISABLED <- function(command, ...) {
...
-    # ===================================================
+    # Perform computations
+
+    # Production shares were computed in a prior file and written out, so just rearrange in one step
+    L1221.ghg_tg_R_agr_C_Y_GLU_IRR <- L122.ghg_tg_R_agr_C_Y_GLU %>%
+      select(-value) %>%
+      repeat_add_columns( tibble(Irr_Rfd = c( "IRR", "RFD" ) ) ) %>%
+      left_join_error_no_match(L1211.ag_irrShare_R_C_Y_GLU_irr, by = c("GCAM_region_ID","GCAM_commodity","GLU","Irr_Rfd","year"))
 
     # Produce outputs
```
Comment (cahartin) [inline R/zchunk_L1221.ghg_agr_R_S_T_Y_IRR.R]: should these comment lines be deleted?

**Comment by rhoesly** [comment]
I think I must have committed the wrong version of the chunk in the original pull request, but the correct tests should run/pass and the issues/comments should be taken care of.

@bpbond should probably check that I've linked to other chunks correctly in the front matter/comment section.

**Comment by bpbond** [comment]
@rhoesly Linking looks fine, thanks. Note you can check that it's good by pulling up the help page (in this case `?module_emissions_L1221.ghg_agr_R_S_T_Y_IRR`) and confirming that links work.

**Comment by bpbond** [comment]
Note you didn't fix the code formatting issues I commented on. I've made those changes.

**Comment by bpbond** [comment]
Holding on @cahartin approval.

File: R/zchunk_L1221.ghg_agr_R_S_T_Y_IRR.R [hunk 7]
```diff
@@ -26,42 +27,30 @@ module_emissions_L1221.ghg_agr_R_S_T_Y_IRR_DISABLED <- function(command, ...) {
...
+
+    # Production shares were computed in a prior file and written out.
+    # Repeat ag emission (L122.ghg_tg_R_agr_C_Y_GLU) for each irrigation (IRR) and rainfed (RFD)
+    # Join in Production share by IRR/RFD then multiply emissions by production share
+    L1221.ghg_tg_R_agr_C_Y_GLU_IRR <- L122.ghg_tg_R_agr_C_Y_GLU %>%
+      repeat_add_columns( tibble(Irr_Rfd = c("IRR", "RFD"))) %>%
+      left_join(L1211.ag_irrShare_R_C_Y_GLU_irr,
+                by = c("GCAM_region_ID", "GCAM_commodity", "GLU", "Irr_Rfd", "year"),
+                suffix = c(".emissions", ".share")) %>% # under timeshift conditions, L122.ghg_tg_R_agr_C_Y_GLU has NaN values (N2O_AGR), which throws error in left_join_no_error as NA values, even though they are matched correctly.
+      mutate(value = value.emissions * value.share) %>%
```
Comment (cahartin) [inline R/zchunk_L1221.ghg_agr_R_S_T_Y_IRR.R]: NaN or NA?

**Pull request merged**